JavaScript basics, Clean code: List style updated according to style guide. (#27275)
TheOdinProject/curriculum: foundations/javascript_basics/clean_code.md

Much better, isn't it? The variable is declared with a descriptive name, and you don't need to perform any calculations when reading this code.

You might wonder why this variable is declared with all caps, when we have recommended camelCase earlier. This is a convention to be used when the programmer is absolutely sure that the variable is *truly* a constant. We know that the milliseconds in an hour will never change, so it's appropriate here.

### Indentation and line length

Now it's time to head to more controversial topics. The war between coders that use tabs and coders that use spaces to indent their code is [essentially a joke by now](https://www.youtube.com/watch?v=SsoOG6ZeyUI).

> 

Reviewer: ```suggestion
```

This comment helps to refresh the reader on how BMI is calculated in plain English, helping the reader to see why the height needs to be converted and what the following calculation is doing. We are almost there just with the naming but the comment still adds further clarity.

### In conclusion

Now that we've covered these ideas, it's good to return to the reminder we shared at the start. Don't try to write perfectly clean code, this will only lead to frustration. Writing "spaghetti" is inevitable, everyone does it sometimes. Just keep these ideas in mind and with time and patience, your code will start to get cleaner.

Learning to write clean code is a process of constant improvement. One that will extend beyond you *completing* The Odin Project. This lesson is meant to serve as a primer, a starting point for that journey.

### Assignment

<div class="lesson-content__panel" markdown="1">

> 1. [This list of clean-code tips](https://onextrapixel.com/10-principles-for-keeping-your-programming-code-clean/).

Reviewer: This isn't on you, it's just an issue we have throughout the curriculum, but when it comes to link text, it's super important that it's descriptive enough. The text should either describe where the link will take a user or its purpose. "This list of clean-code tips" isn't terrible, but it can still be improved:

```suggestion
1. Read [10 Principles for Keeping Your Programming Code Clean](https://onextrapixel.com/10-principles-for-keeping-your-programming-code-clean/) to get some great tips for clean code.
```

> 1. [This article](https://blog.codinghorror.com/coding-without-comments/), [and this one too](https://blog.codinghorror.com/code-tells-you-how-comments-tell-you-why/) about the role of comments in your code.

Reviewer: ```suggestion
1. To help better understand good comment practices, read about [comments telling us how code works](https://blog.codinghorror.com/code-tells-you-how-comments-tell-you-why/) as well as how to [code without comments](https://blog.codinghorror.com/coding-without-comments/).
```

> 1. To help better understand good comment practices, read about [comments telling us how code works](https://blog.codinghorror.com/code-tells-you-how-comments-tell-you-why/) as well as how to [code without comments](https://blog.codinghorror.com/coding-without-comments/).

Reviewer: ```suggestion
1. To help better understand good comment practices, read about [comments telling us how code works](https://blog.codinghorror.com/code-tells-you-how-comments-tell-you-why/) as well as how to [code without comments](https://blog.codinghorror.com/coding-without-comments/).

```

</div>

### Knowledge check

> This section contains questions for you to check your understanding of this lesson on your own. If you’re having trouble answering a question, click it and review the material it links to.

Reviewer: ```suggestion
The following questions are an opportunity to reflect on key topics in this lesson. If you can't answer a question, click on it to review the material, but keep in mind you are not expected to memorize or master this knowledge.
```

- [A nice op-ed](https://www.martinfowler.com/bliki/CodeAsDocumentation.html)
- [Airbnb style guide](https://github.com/airbnb/javascript)
- [Chaining methods to write sentences](https://web.archive.org/web/20190211152543/https://javascriptissexy.com/beautiful-javascript-easily-create-chainable-cascading-methods-for-expressiveness/)

> - [Clean code in Javascript](https://github.com/ryanmcdermott/clean-code-javascript)

Reviewer: ```suggestion
- [Clean code in JavaScript](https://github.com/ryanmcdermott/clean-code-javascript)
```

> 
> - [Clean code in JavaScript](https://github.com/ryanmcdermott/clean-code-javascript)

Reviewer: ```suggestion
- [Clean code in JavaScript](https://github.com/ryanmcdermott/clean-code-javascript)
```
